### DISCUSSION TITLE
Column Helper Accessor Function Update on Component State Updates

### DISCUSSION THREAD

**Discussion by akibmayadav** [TanStack/table#4868] [Q&A]
Hi, 

I am using accessor Function to format my cell (to make sure filtering works with formatted Values). 
In my use case, the user can determine the custom formatting rules. In this example : convert last name cell value to uppercase. 

The formatting rules are component state, but when they update the accessor function does not update. As a result the cell's do not reflect the formatting changes. 

Here is my sandbox: https://codesandbox.io/p/sandbox/interesting-parm-z247ye?welcome=true. 

Am I missing something here ?

**Comment by AdmiJW** [comment]
Hi, I am also having this problem as well. In an attempt to solve it, I have found a workaround:

It seems like the accessor function will only be updated if `data` is also updated. The `data` is never changed in your sandbox, so the accessor function will only be run once, not when the `isUpperCase` changes.

My workaround is, whenever the dependencies change, which is `isUpperCase` in your case, return a different copy of the `data` which contains the same content using `useMemo`. This way the `data` will be perceived as changed, and the accessor function will be updated accordingly.

```ts
const rows = React.useMemo(() => [...data], [data, isUpperCase]);
```

Here's the sandbox with my workaround applied: [HERE](https://codesandbox.io/p/devbox/zealous-glade-zvjjr7?file=%2Fsrc%2Fmain.tsx&workspaceId=1c3664b4-15e6-4a7f-95a6-e8e9d5c01266)


This is also mentioned in the [docs](https://tanstack.com/table/latest/docs/api/core/table#data).

> When the `data` option changes reference (compared via `Object.is`), the table will reprocess the data. Any other data processing that relies on the core data model (such as grouping, sorting, filtering, etc) will also be reprocessed.

>  Make sure your `data` option is only changing when you want the table to reprocess. Providing an inline `[]` or constructing the `data` array as a new object every time you want to render the table will result in a lot of unnecessary re-processing. This can easily go unnoticed in smaller tables, but you will likely notice it in larger tables.
